File: src/main/java/com/pl/virussimulation/simulation/Simulation.java
package com.pl.virussimulation.simulation;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "players",
        uniqueConstraints = {
                @UniqueConstraint(columnNames = "nickname")})
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Simulation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;
    String n; //simulation name
    Integer p; //population quantity
    Integer i; //initial quantity of sick
    Integer r; //infection rate
    Integer m; //death rate
    Integer ti; //avg days to recover
    Integer tm; //avg days to die
    Integer ts; // time of simulation

}

File: src/frontend/Angular/src/app/shared/simulation.ts
export interface Simulation {
    n:String; //simulation name
    p:number; //population quantity
    i:number; //initial quantity of sick
    r:number; //infection rate
    m:number; //death rate
    ti:number; //avg days to recover
    tm:number; //avg days to die
    ts:number ; // time of simulation

}
